tests/components/application-connector/test/compass-runtime-agent/synchronisation_test.go

```diff
@@ -65,3 +73,39 @@ func (cs *CompassRuntimeAgentSuite) assignApplicationToFormationAndWaitForSync(a
 		Timeout:              appCreationTimeout,
 	}.Do()
 }
+
+func (cs *CompassRuntimeAgentSuite) removeApplicationAndWaitForSync(appReader ApplicationReader, compassAppName, applicationID string) error {
+	t := cs.T()
+	t.Helper()
+
+	exec := func() error {
+		err := cs.directorClient.UnassignApplication(applicationID, cs.formationName)
+		if err != nil {
+			return err
+		}
+
+		err = cs.directorClient.UnregisterApplication(applicationID)
+		return err
+	}
+
+	verify := func() bool {
+		_, err := appReader.Get(context.Background(), compassAppName, v1.GetOptions{})
+		if err != nil {
+			var statusErr *k8sErr.StatusError
+			if errors.As(err, &statusErr) && statusErr.Status().Reason == v1.StatusReasonNotFound {
+				t.Logf("App not found after removing (this is ok): %v", err)
+				return true
+			}
+			t.Logf("Failed to get app after removing (this is bad): %v", err)
```

Comment:
@akgalwas noticed that these messages could be changed to some like `"Application was successfully removed" and "Failed to check whether Application was removed"` instead of `(this is ok/ bad)`